.github/workflows/ci.yml

```diff
@@ -27,9 +33,24 @@ jobs:
       - run: npm run prepublishOnly
 
       - uses: primer/publish@v3.0.0
+        id: publish
         with:
-          npm_args: --unsafe-perm --allow-same-version
+          npm_args: --unsafe-perm
           default_branch: main
         env:
           GITHUB_TOKEN: ${{ github.token }}
-          NPM_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }}
+
+      - name: set BRANCH_DIST_TAG
+        if: github.ref_type == 'branch'
```

Comment:
i'm an Actions newb, so apologies, but I'm guessing this line here makes sure this won't run on merge commits into `main`? assuming we don't want a branch.main along with the actual published version 😅